File: app/src/main/java/com/booknara/android/apps/patterns/fragment/SimpleFragment.java
package com.booknara.android.apps.patterns.fragment;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.booknara.android.apps.patterns.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class SimpleFragment extends Fragment {
    public static final String TAG = "SimpleFragment";
    private static final String BODY = "BODY";

    private TextView mBodyText;
    private String mBody;

    public static Fragment newInstance() {
        return new SimpleFragment();
    }

    public static Fragment newInstance(String body) {
        Bundle bundle = new Bundle();
        bundle.putString(BODY, body);
        Fragment fragment = new SimpleFragment();
        fragment.setArguments(bundle);

        return fragment;
    }
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle bundle = getArguments();
        if (bundle != null) {
            mBody = bundle.getString(BODY);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_simple, container, false);
        mBodyText = view.findViewById(R.id.simple_text);

        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (!TextUtils.isEmpty(mBody)) {
            mBodyText.setText(mBody);
        }
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onStop() {
        super.onStop();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/booknara/android/apps/patterns/fragment/SimpleFragment.kt
package com.booknara.android.apps.patterns.fragment

import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import androidx.fragment.app.Fragment
import com.booknara.android.apps.patterns.R
import com.booknara.android.apps.patterns.databinding.FragmentSimpleBinding

class SimpleFragment: Fragment(R.layout.fragment_simple) {

  private lateinit var binding: FragmentSimpleBinding
  private lateinit var text: String

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    text = arguments?.getString(BODY) ?: ""
    Log.i(TAG, "text: $text")
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    binding = FragmentSimpleBinding.bind(view)
  }

  override fun onActivityCreated(savedInstanceState: Bundle?) {
    super.onActivityCreated(savedInstanceState)
    if (!TextUtils.isEmpty(text)) {
      binding.simpleText.text = text
    }
  }

  companion object {
    const val TAG = "SimpleFragment"
    const val BODY = "BODY"

    @JvmStatic
    fun newInstance(): Fragment {
      return SimpleFragment()
    }

    @JvmStatic
    fun newInstance(body: String): Fragment {
      val bundle = Bundle()
      bundle.putString(BODY, body)
      val fragment = SimpleFragment()
      fragment.arguments = bundle
      return fragment
    }
  }
}
